tests/bgp/test_passive_peering.py

```diff
@@ -0,0 +1,215 @@
+'''
+
+This script is to test BGP passive peering on SONiC.
+
+'''
+
+import logging
+import pytest
+import time
+from tests.common.config_reload import config_reload
+from tests.common.helpers.constants import DEFAULT_NAMESPACE
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+bgp_config_sleeptime = 90
+peer_password = "sonic.123"
+wrong_password = "wrong-password"
+
+
+@pytest.fixture(scope='module')
+def setup(tbinfo, nbrhosts, duthosts, rand_one_dut_hostname, request):
+    # verify neighbors are type sonic
+    isSonic = False
+    if request.config.getoption("neighbor_type") == "sonic":
+        isSonic = True
```

Comment:
could you do pytest_skip as in the original code, for non-sonic? iso using this flag

Reply:
@wenyiz2021 Yes, I changed to using skip now.